numpy/__init__.pyi

```diff
@@ -386,133 +388,172 @@ class generic(_ArrayOrScalarCommon):
+        self, __value: Union[SupportsInt, SupportsIndex, str, bytes] = ...
+    ) -> None: ...
 
 class uint32(unsignedinteger):
-    def __init__(self, __value: SupportsInt = ...) -> None: ...
+    def __init__(
+        self, __value: Union[SupportsInt, SupportsIndex, str, bytes] = ...
+    ) -> None: ...
 
 class uint64(unsignedinteger):
-    def __init__(self, __value: SupportsInt = ...) -> None: ...
+    def __init__(
+        self, __value: Union[SupportsInt, SupportsIndex, str, bytes] = ...
+    ) -> None: ...
 
 class inexact(number): ...  # type: ignore
-class floating(inexact, _real_generic): ...  # type: ignore
+class floating(inexact): ...  # type: ignore
 
 _FloatType = TypeVar('_FloatType', bound=floating)
 
 class float16(floating):
-    def __init__(self, __value: Optional[SupportsFloat] = ...) -> None: ...
+    def __init__(
+        self, __value: Union[None, str, bytes, SupportsFloat, SupportsIndex] = ...
+    ) -> None: ...
 
 class float32(floating):
-    def __init__(self, __value: Optional[SupportsFloat] = ...) -> None: ...
+    def __init__(
+        self, __value: Union[None, str, bytes, SupportsFloat, SupportsIndex] = ...
+    ) -> None: ...
 
-class float64(floating):
-    def __init__(self, __value: Optional[SupportsFloat] = ...) -> None: ...
+class float64(floating, float):
+    def __init__(
+        self, __value: Union[None, str, bytes, SupportsFloat, SupportsIndex] = ...
+    ) -> None: ...
 
 class complexfloating(inexact, Generic[_FloatType]):  # type: ignore
     @property
-    def real(self) -> _FloatType: ...
+    def real(self) -> _FloatType: ...  # type: ignore[override]
     @property
-    def imag(self) -> _FloatType: ...
+    def imag(self) -> _FloatType: ...  # type: ignore[override]
     def __abs__(self) -> _FloatType: ...  # type: ignore[override]
 
 class complex64(complexfloating[float32]):
     def __init__(
         self,
-        __value: Union[None, SupportsInt, SupportsFloat, SupportsComplex] = ...
+        __value: Union[
+            None, str, bytes, SupportsFloat, SupportsComplex, SupportsIndex
+        ] = ...,
     ) -> None: ...
 
-class complex128(complexfloating[float64]):
+class complex128(complexfloating[float64], complex):
     def __init__(
         self,
-        __value: Union[None, SupportsInt, SupportsFloat, SupportsComplex] = ...
+        __value: Union[
+            None, str, bytes, SupportsFloat, SupportsComplex, SupportsIndex
+        ] = ...,
     ) -> None: ...
 
-class flexible(_real_generic): ...  # type: ignore
+class flexible(generic): ...  # type: ignore
 
 class void(flexible):
-    def __init__(self, __value: Union[int, integer, bool_, bytes, bytes_]): ...
+    def __init__(self, __value: Union[int, integer, bool_, bytes]): ...
+    @property
+    def real(self: _ArraySelf) -> _ArraySelf: ...
+    @property
+    def imag(self: _ArraySelf) -> _ArraySelf: ...
 
-class character(_real_generic): ...  # type: ignore
+class character(generic): ...  # type: ignore
```

Comment:
```suggestion
class character(flexible): ...  # type: ignore
```

Reply:
Fixed in https://github.com/numpy/numpy/pull/17214/commits/b86a04f81d27a2fbefe574db069a023113b373cf.